### ISSUE TITLE
inveniogc doesn't follow symlinks

### ISSUE THREAD

**Issue by tsgit** [inspirehep/invenio#370]
see discussion at https://github.com/inspirehep/invenio/pull/311

if `CFG_SOMEDIR` is a symlink then `find CFG_SOMEDIR` does not work as intended. This affects all inveniogc rules which rely on `find CFG_SOMEDIR ...` unless `CFG_SOMEDIR` ends in `/.`

proposed solution see https://github.com/inspirehep/invenio/pull/311

**Comment by tsgit** [comment]
so if symlinks should be followed I suggest to check all path variables and replace them with `realpath` at the beginning of `inveniogc`

```
In [8]: from invenio.config import CFG_TMPDIR

In [9]: from os.path import (islink, realpath)

In [10]: CFG_TMPDIR
'/opt/cds-invenio/var/tmp'

In [11]: if islink(CFG_TMPDIR):
   ....:     CFG_TMPDIR = realpath(CFG_TMPDIR)
   ....:     

In [12]: CFG_TMPDIR
'/data01/tmp'
```

**Comment by kaplun** [comment]
That sounds a good approach to me. In this way you don't have to blindly follow symlinks.

**Comment by tsgit** [comment]
I overlooked that this was addressed already in 72e9879c

https://github.com/inspirehep/invenio/blob/prod/modules/websession/lib/inveniogc.py#L48-L51

closing the issue. sorry for the noise.
